contracts/FastBreak.cdc

```diff
@@ -0,0 +1,682 @@
+/*
+    Fast Break Game Contract
+    Author: Jeremy Ahrens jer.ahrens@dapperlabs.com
+*/
+
+import NonFungibleToken from 0xNFTADDRESS
+import TopShot from 0xTOPSHOTADDRESS
+
+/// Game & Oracle Contract for Fast Break
+///
+pub contract FastBreak: NonFungibleToken {
+
+    /// Contract events
+    ///
+    pub event ContractInitialized()
+    pub event Withdraw(id: UInt64, from: Address?)
+    pub event Deposit(id: UInt64, to: Address?)
+    pub event FastBreakRunCreated(
+        id: String,
+        name: String,
+        runStart: UInt64,
+        runEnd: UInt64,
+        fatigueModeOn: Bool
+    )
+    pub event FastBreakRunStatusChange(id: String, newStatus: String)
+    pub event FastBreakGameCreated(
+        id: String,
+        name: String,
+        fastBreakRunID: String,
+        isPublic: Bool,
+        submissionDeadline: UInt64,
+        numPlayers: UInt64
+    )
+    pub event FastBreakGameStatusChange(id: String, newStatus: String)
+    pub event FastBreakNFTBurned(id: UInt64, serialNumber: UInt64)
+    pub event FastBreakNFTMinted(
+        id: UInt64,
+        fastBreakGameID: String,
+        serialNumber: UInt64,
+        mintingDate: UInt64,
+        topShots: [UInt64],
+        mintedTo: Address
+    )
+    pub event FastBreakSubmissionSent(
+        wallet: Address,
+        submittedAt: UInt64,
+        fastBreakGameID: String,
+        topShots: [UInt64]
+    )
+    pub event FastBreakGameWinner(
+        wallet: Address?,
+        submittedAt: UInt64,
+        fastBreakGameID: String,
+        topShots: [UInt64]
+    )
+    pub event FastBreakGameStatAdded(
+        fastBreakGameID: String,
+        name: String,
+        type: String,
+        valueNeeded: UInt64
+    )
+
+    /// Named Paths
+    ///
+    pub let CollectionStoragePath:  StoragePath
+    pub let CollectionPublicPath:   PublicPath
+    pub let OracleStoragePath:       StoragePath
+    pub let OraclePrivatePath:      PrivatePath
+
+    /// Contract variables
+    ///
+    pub var totalSupply:        UInt64
+
+    /// Metadata Dictionaries
+    ///
+    access(self) let fastBreakRunByID:        {String: FastBreakRun}
+    access(self) let fastBreakGameByID:           {String: FastBreakGame}
+
+    /// A top-level Fast Break Run, the container for Fast Break Games
```

Comment:
What do you mean by "container for fast break games"

Reply:
Will make the comment better, but a Fast Break occurs on any day there are NBA games. A Fast Break Run is a collection of Fast Breaks spanning about 2 weeks. We have rewards setup for hitting thresholds. 

<img width="458" alt="Screen Shot 2023-12-21 at 12 18 59 PM" src="https://github.com/dapperlabs/nba-smart-contracts/assets/896651/b0c11549-83da-40e4-8b82-7c779b711b5b">